agents/aliyun/fence_aliyun.py

```diff
@@ -58,20 +72,22 @@ def get_status(conn, instance_id):
 	return instance_status
 
 def get_nodes_list(conn, options):
+	logging.debug("start to get nodes list")
 	result = {}
 	request = DescribeInstancesRequest()
 	request.set_PageSize(100)
 	response = _send_request(conn, request)
-	instance_status = None
 	if response is not None:
 		instance_list = response.get('Instances').get('Instance')
 		for item in instance_list:
 			instance_id = item.get('InstanceId')
-			instance_name = item.get('InstanceName')
-			result[instance_id] = (instance_name, None)
+			host_name = item.get('HostName')
```

Comment:
I think we'll have to do this as a try/except (EncodingError or whatever the exception it throws is) to avoid breaking functionality for current users who might not have same instance name and host name.

Reply:
Sorry, I’m wondering,  what is the impact of users using different instance name and host name?

Reply:
It will stop working if they got it setup with instance name and they upgrade and all of a sudden it doesnt work with instance names anymore.

Reply:
Thank you for your answer, I think I understand what you mean, but I have another question: how does the instance_name  cause the user to stop working before and after the upgrade?

In addition, this API will definitely return hostname after the upgrade, and there is no failure to obtain it.

Reply:
Multiple nodes could have the same hostname. E.g. if you have instance name cluster1-node1 with hostname node1 and instance cluster2-node1 with node1 as hostname (probably different users using it in that case)

Reply:
Yes, the same hostname is possible.
But I think in this scenario `item.get('HostName')` will not raise an exception, it will only generate a map like this:
`{"instance-id-1": ("code1", None), "instance-id-2": ("code1", None)}`, try/except is not working.

Or will such a map containing the same hostname cause errors in subsequent executions? (If yes, can you provide more details, thanks)

Reply:
Is it possible to use item.get('InstanceId').decode() or .encode() to solve the issue?

If not we should make a parameter called name_lookup or similar that defaults to instance and can be changed to hostname.

And we can also do the .get() in a try/except that informs users about the parameter if it fails.

Reply:
Maybe I didn’t describe clearly the problem we encountered, I have restored the code here and created an issue (https://github.com/ClusterLabs/fence-agents/issues/452#issue-1053209050)
If necessary, I will solve this problem through a new PR.